core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1424,6 +1470,28 @@ private[spark] class BlockManager(
     blockStoreUpdater.save()
   }
 
+  // Check whether a rdd block is visible or not.
+  private[spark] def isRDDBlockVisible(blockId: RDDBlockId): Boolean = {
+    // Cached blocks are always visible if the feature flag is disabled.
+    if (!trackingCacheVisibility) {
+      return true
+    }
+
+    // If the rdd block visibility information not available in the block manager,
+    // asking master for the information.
+    if (blockInfoManager.isRDDBlockVisible(blockId)) {
+      return true
+    }
+
+    if(master.isRDDBlockVisible(blockId)) {
+      // Cache the visibility status if block exists.
+      blockInfoManager.tryAddVisibleBlock(blockId)
+      true
```

Comment:
Should we still return `true` if `blockInfoManager` doesn't contain the block?

Reply:
Yes, I think so. Even though current executor doesn't have the cached block, we still can read the cache from a remote executor.